sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -79,6 +81,49 @@ export class Ed25519Keypair implements Keypair {
     return new Ed25519Keypair(nacl.sign.keyPair.fromSeed(seed));
   }
 
+  /**
+   * Test Derive Path from path string
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
+   */
+  static isValidPath = (path: string): boolean => {
+    if (
+      !new RegExp("^m\\/44+'\\/784+'\\/[0-9]+'\\/[0-9]+'\\/[0-9]+'+$").test(
+        path
+      )
+    ) {
+      return false;
+    }
+    return !path
+      .split('/')
+      .slice(1)
+      .map((val: string): string => val.replace("'", ''))
+      .some(isNaN as any);
+  };
+
+  /**
+   * Generate a keypair from mnemonics.
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
```

Comment:
any advantage to expose the path as a param? maybe worth hardcode as a config?

Reply:
The HDPaths of some blockchains did not have a clear path defined initially, so the addresses created in the HW Ledger javascript SDK were different.

I think you should go strictly to avoid such confusion.

![solana](https://user-images.githubusercontent.com/57783762/179661977-a39e97d3-1159-4194-a589-1a7985d883d9.png)
.